File: 205_isomorphic.py
class Solution:
    def wordPattern(self, s: str, t: str) -> bool:
        if len(s) != len(t): 
            return False
        hash_map = {}
        for i in range(len(s)):
            if  s[i] not in hash_map:
                if  t[i] in hash_map.values():
                    return False
                hash_map[s[i]]=t[i]               
            else:
                if hash_map[s[i]]!=t[i]:
                    return False 
        return True

s = "egg"
t = "add"
print(Solution().wordPattern(s, t)) 

File: 205_isomorphic.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        map<char,char> hash;
        if(s.size()!=t.size()) return false;
        int i= 0;
        for(auto c : s){
            if()
            hash[c]=t[i];
            i++;

        }
    }
};

int main(){

    Solution sol;
    cout<<sol.isIsomorphic("egg", "add")<<endl;
    return 0;
}